### ISSUE TITLE
Opening/Remuxing TS to MKV fails

### ISSUE THREAD

**Issue by gnucian** [staxrip/staxrip#149]
When a TS file has an AAC audio stream (in ADTS muxing mode), StaxRip fails to open/remux it to an MKV format via FFmpeg.

Here's the log: [error log](https://pastebin.com/fhB96S8K)
You can get the source video file from [here](https://4kmedia.org/lg-new-york-hdr-uhd-4k-demo/).

The error arises because StaxRip doesn't employ the proper FFmpeg input options and bitstream filter required to remux an ADTS audio stream.

- input options: `-analyzeduration 100M -probesize 50M`
- bitstream filter: `-bsf:a aac_adtstoasc`

Fixing the ffmpeg command line as follows, I was able to successfully remux the same TS file:

`ffmpeg -hide_banner -analyzeduration 100M -probesize 50M -i "LG New York HDR UHD 4K Demo.ts" -map 0 -c copy -bsf:a aac_adtstoasc -y "LG New York HDR UHD 4K Demo.mkv"`

Thanks.

**Comment by 44vince44** [comment]
@gnucian Please can you try the following thing, it may help find the reason:
-> rchange the extension of your file from ts to flv.
And try to open the file. Does it open?

**Comment by gnucian** [comment]
Nope. Same error.

**Comment by 44vince44** [comment]
In the audio encoder windows, is **"decoder"** set to "Automatic" ?

**Comment by gnucian** [comment]
Yes.

Per your suggestion, I tried changing it to **ffmpeg** and opening the ts file, but still the same error occurs.

I believe that it is not a matter of decoding, but a matter of muxing method.

**Comment by 44vince44** [comment]
I was trying to see whether your problem matches a similar problem found by another user.
Well it seems not...
I hope Stax will take a look at it

**Comment by 44vince44** [comment]
BTW, when you opened this issue several days ago, I tried to download the video from the link, and I was forbidden the download due to many people downloaded that video. I just tried again, right now, and I couldn't either, for the same reason...

**Comment by gnucian** [comment]
> BTW, when you opened this issue several days ago, I tried to download the video from the link, and I was forbidden the download due to many people downloaded that video. I just tried again, right now, and I couldn't either, for the same reason... Tried downloading another file from the site, same error!

That's very weird. I never failed to download the video file, and I'm still able to download the file.

Anyway, I've uploaded it to my Google drive: [here](https://drive.google.com/open?id=1iuQcLkMMdKYM_KzFBTvaNucDtbviGdHZ). It will be deleted after a 72 hours.

**Comment by stax76** [comment]
You can modify the pre processing at:

Main Menu > Tool > Settings > Pre Processing

Modify the existing step 'ffmpeg: Re-mux TS to MKV'

Command: `powershell.exe`
Arguments: `-nologo -file "D:\ffmpegTsToMkv.ps1" "%source_file%" "%temp_file%.mkv"`

Code:

```PowerShell
$muxingMode = Get-MediaInfoValue ($args[0]) -Kind Audio -Parameter MuxingMode

if ($muxingMode -eq 'ADTS')
{
    $bsf = '-bsf:a', 'aac_adtstoasc'
}

ffmpeg -hide_banner -analyzeduration 100M -probesize 50M -i $args[0] -map 0 -c copy @bsf -y $args[1]
```

You need to install Get-MediaInfo however and add ffmpeg to the path env var or use the build from today, it adds all exe tools to the path env var and adds all macros as env vars.

**Comment by gnucian** [comment]
Thanks.

I did some modification on your suggestion as follows and it works smoothly.

1) I dot sourced Get-MediaInfo included in StaxRip in the PowerShell $PROFILE file, but since Get-MediaInfo in StaxRip does not have MediaInfo.dll in the same path, **I had to copy MediaInfo.dll into the Get-MediaInfo folder in order to avoid extra errors**.

2) I put ffmpegTsToMkv.ps1 in the Get-MediaInfo folder and changed the pre-processing arguments as follows:

`-nologo -file "%app_dir:Get-MediaInfo%ffmpegTsToMkv.ps1" "%source_file%" "%temp_file%.mkv"`

※ ffmpeg is already in my PATH, so it didn't matter.

Thanks again.
